### ISSUE TITLE
BLIP-2: Bootstrapping Language-Image Pre-training with Frozen Image Encoders and Large Language Models

### ISSUE THREAD

**Issue by mindyjkang** [mindyjkang/papers-i-read#2]
# 📌 Paper Information

 * Title: BLIP-2: Bootstrapping Language-Image Pre-training with Frozen Image Encoders and Large Language Models
 * Authors: Junnan Li Dongxu Li Silvio Savarese Steven Hoi
* Year: 2023
* Link: [https://arxiv.org/abs/2301.12597](https://arxiv.org/abs/2301.12597)
* Keywords: Zero-shot image-to-text generation, Visual Question Answering, Image Captioning, Multimodals

# 📝 Summary
 
BLIP-2 trains Vision-Language model by leveraging frozen pretrained image encoders & LLMs.

## Querying Transformer (Q-Former)

![Image](https://github.com/user-attachments/assets/74abc93d-70cc-4d03-81f1-7831520a12f2)

* Incorporates learnable query vectors to extract visual features from frozen image encoder
* Consisted of 2 Transformer modules; Image Transformer & Text Transformer

## BLIP-2: First Training Stage

![Image](https://github.com/user-attachments/assets/814b803a-f7f6-426b-b7e5-811c16b7661d)

* Vision-Language representation learning
* Train Q-Formers to learn visual representation most relevant to the text

### Image-Text Contrastive Learning

* learns to align image representation and text representation 
* constrasting the image-text similarity of a positive pair against those of negative pairs
* Unimodal self-attention mask : queries and text are not allowed to see each other

### Image-grounded Text Generation

* Train Q-Former to generate texts given input images as the condition
* Learnable queires are trained to extract visual features that capture all information about the text
* Multimodal causal self-attention mask : 
   * queries can attend to each other but not the text tokens
   * each text token can attend to all queries and its previous text tokens

### Image-Text Matching

* binary classification task : image-text pair is positive or negative
* Bi-directional self-attention mask : all queries and texts can attend to each other
* feed each output query embedding into a two-class linear classifier

## BLIP-2: Second Training Stage

![Image](https://github.com/user-attachments/assets/d9fdde48-eef2-465d-ac81-b7f36ffa0267)

* Vision-to-language generation learning
* Trains Q-Former such that its output visual representation can be interpreted by the LLM

1. Connect Q-Former to a frozen LLM
2. Prepend query embedding to input text embedding of LLM
